File: MultiplicacionMatrices.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */

/**
 *
 * @author Monica Ileana
 */
import java.util.Scanner;

public class MultiplicacionMatrices {
    
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        
        System.out.println("=== MULTIPLICACIÓN DE MATRICES ===");
        System.out.print("Filas de la matriz A: ");
        int filasA = scanner.nextInt();
        System.out.print("Columnas de la matriz A: ");
        int columnasA = scanner.nextInt();
        
        System.out.print("Filas de la matriz B: ");
        int filasB = scanner.nextInt();
        System.out.print("Columnas de la matriz B: ");
        int columnasB = scanner.nextInt();
        
        if (columnasA != filasB) {
            System.out.println("Error: No se pueden multiplicar las matrices.");
            System.out.println("El número de columnas de A debe ser igual al número de filas de B.");
            scanner.close();
            return;
        }
        
        int[][] matrizA = new int[filasA][columnasA];
        int[][] matrizB = new int[filasB][columnasB];
        int[][] matrizResultado = new int[filasA][columnasB];
        
        System.out.println("\nIngrese los elementos de la matriz A:");
        for (int i = 0; i < filasA; i++) {
            for (int j = 0; j < columnasA; j++) {
                System.out.print("A[" + i + "][" + j + "]: ");
                matrizA[i][j] = scanner.nextInt();
            }
        }
        
        System.out.println("\nIngrese los elementos de la matriz B:");
        for (int i = 0; i < filasB; i++) {
            for (int j = 0; j < columnasB; j++) {
                System.out.print("B[" + i + "][" + j + "]: ");
                matrizB[i][j] = scanner.nextInt();
            }
        }
        
        multiplicarMatrices(matrizA, matrizB, matrizResultado);
        

        System.out.println("\n=== RESULTADOS ===");
        
        System.out.println("Matriz A:");
        mostrarMatriz(matrizA);
        
        System.out.println("Matriz B:");
        mostrarMatriz(matrizB);
        
        System.out.println("Matriz Resultado (A × B):");
        mostrarMatriz(matrizResultado);
        
        scanner.close();
    }
    
    public static void multiplicarMatrices(int[][] A, int[][] B, int[][] resultado) {
        int filasA = A.length;
        int columnasA = A[0].length;
        int columnasB = B[0].length;
        
        for (int i = 0; i < filasA; i++) {
            for (int j = 0; j < columnasB; j++) {
                int suma = 0;
                for (int k = 0; k < columnasA; k++) {
                    suma += A[i][k] * B[k][j];
                }
                resultado[i][j] = suma;
            }
        }
    }
    
    public static void mostrarMatriz(int[][] matriz) {
        for (int i = 0; i < matriz.length; i++) {
            for (int j = 0; j < matriz[0].length; j++) {
                System.out.print(matriz[i][j] + "\t");
            }
            System.out.println();
        }
        System.out.println();
    }
}

File: MultiplicacionMatrices.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int filasA, columnasA, filasB, columnasB;
    
    printf("=== MULTIPLICACIÓN DE MATRICES ===\n");
    printf("Filas de la matriz A: ");
    scanf("%d", &filasA);
    printf("Columnas de la matriz A: ");
    scanf("%d", &columnasA);
    
    printf("Filas de la matriz B: ");
    scanf("%d", &filasB);
    printf("Columnas de la matriz B: ");
    scanf("%d", &columnasB);
    
    if (columnasA != filasB) {
        printf("Error: No se pueden multiplicar las matrices.\n");
        printf("El número de columnas de A debe ser igual al número de filas de B.\n");
        return 1;
    }
    
    int **matrizA = (int **)malloc(filasA * sizeof(int *));
    int **matrizB = (int **)malloc(filasB * sizeof(int *));
    int **matrizResultado = (int **)malloc(filasA * sizeof(int *));
    
    for (int i = 0; i < filasA; i++) {
        matrizA[i] = (int *)malloc(columnasA * sizeof(int));
        matrizResultado[i] = (int *)malloc(columnasB * sizeof(int));
    }
    
    for (int i = 0; i < filasB; i++) {
        matrizB[i] = (int *)malloc(columnasB * sizeof(int));
    }
    
    printf("\nIngrese los elementos de la matriz A:\n");
    for (int i = 0; i < filasA; i++) {
        for (int j = 0; j < columnasA; j++) {
            printf("A[%d][%d]: ", i, j);
            scanf("%d", &matrizA[i][j]);
        }
    }
    
    printf("\nIngrese los elementos de la matriz B:\n");
    for (int i = 0; i < filasB; i++) {
        for (int j = 0; j < columnasB; j++) {
            printf("B[%d][%d]: ", i, j);
            scanf("%d", &matrizB[i][j]);
        }
    }
    
    for (int i = 0; i < filasA; i++) {
        for (int j = 0; j < columnasB; j++) {
            matrizResultado[i][j] = 0;
            for (int k = 0; k < columnasA; k++) {
                matrizResultado[i][j] += matrizA[i][k] * matrizB[k][j];
            }
        }
    }
    
    printf("\n=== RESULTADOS ===\n");
    
    printf("Matriz A:\n");
    for (int i = 0; i < filasA; i++) {
        for (int j = 0; j < columnasA; j++) {
            printf("%4d ", matrizA[i][j]);
        }
        printf("\n");
    }
    printf("\n");
    
    printf("Matriz B:\n");
    for (int i = 0; i < filasB; i++) {
        for (int j = 0; j < columnasB; j++) {
            printf("%4d ", matrizB[i][j]);
        }
        printf("\n");
    }
    printf("\n");
    
    printf("Matriz Resultado (A × B):\n");
    for (int i = 0; i < filasA; i++) {
        for (int j = 0; j < columnasB; j++) {
            printf("%6d ", matrizResultado[i][j]);
        }
        printf("\n");
    }
    
    for (int i = 0; i < filasA; i++) {
        free(matrizA[i]);
        free(matrizResultado[i]);
    }
    for (int i = 0; i < filasB; i++) {
        free(matrizB[i]);
    }
    free(matrizA);
    free(matrizB);
    free(matrizResultado);
    
    return 0;
}